app/pipes.go

```diff
@@ -25,6 +29,18 @@ func RegisterPipeRoutes(router *mux.Router, pr PipeRouter) {
 		HandlerFunc(requestContextDecorator(deletePipe(pr)))
 }
 
+func checkPipe(pr PipeRouter, end End) CtxHandlerFunc {
+	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) {
+		id := mux.Vars(r)["pipeID"]
+		_, _, ok := pr.Get(ctx, id, end)
+		if !ok {
+			http.NotFound(w, r)
+			return
+		}
+		defer pr.Release(ctx, id, end)
```

Comment:
The `defer` is a little redundant.

Also might want to do `w.WriteHeader(http.StatusNoContent)`